rats-rs/src/transport/tls/client.rs

```diff
@@ -37,14 +37,17 @@ pub struct Client {
     attest_self: bool,
 }
 
+unsafe impl Send for Client {}
```

Comment:
I feel a bit confusing, since I am not who actually writing this code and fighting with the rustc compiler. But could you explain the reason for adding `Send` here? 😂

Reply:
Sorry for this confusion. The reason for unsafe implementing the `Send` trait is that when you enable both the `transport-tls` and `async-tokio` features (of course, you need to disable the `is-async` feature), a compilation error related to async will occur.

```
error: future cannot be sent between threads safely
   --> rats-rs/src/transport/tls/client.rs:97:58
    |
97  |       async fn send(&mut self, bytes: &[u8]) -> Result<()> {
    |  __________________________________________________________^
98  | |         if self.ssl_session.is_none() {
99  | |             return Err(Error::kind(ErrorKind::OsslCtxOrSessionUninitialized));
100 | |         }
...   |
111 | |         Ok(())
112 | |     }
    | |_____^ future created by async block is not `Send`
    |
    = help: within `{async block@rats-rs/src/transport/tls/client.rs:97:58: 112:6}`, the trait `Send` is not implemented for `*mut ssl_st`, which is required by `{async block@rats-rs/src/transport/tls/client.rs:97:58: 112:6}: Send`
note: captured value is not `Send` because `&mut` references cannot be sent unless their referent is `Send`
   --> rats-rs/src/transport/tls/client.rs:97:24
    |
97  |     async fn send(&mut self, bytes: &[u8]) -> Result<()> {
    |                        ^^^^ has type `&mut Client` which is not `Send`, because `Client` is not `Send`
    = note: required for the cast from `Pin<Box<{async block@rats-rs/src/transport/tls/client.rs:97:58: 112:6}>>` to `Pin<Box<(dyn Future<Output = std::result::Result<(), errors::Error>> + Send + 'async_trait)>>`
```

In brief, this error says that Future can't auto impl `Send` because `&mut Self` is not `Send`. The reason why `Self` is not `Send` is that it contains a raw pointer which is not `Send`. This raw pointer is actually managed by `openssl-sys` lib(on earth openssl lib), so the easiest way to bypass this error is unsafe impl `Send` for `Client` and `Server`.

I will push a new fix commit to resolve this confusion:
1. add conditional compilation for this unsafe impl with comment
2. restructure features in crates(current rats-rs features doesn't support enable `transport-tls` and `async-tokio` simultaneously because of my mistake)